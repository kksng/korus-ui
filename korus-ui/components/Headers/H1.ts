import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H1 = htmlTagFactory('H1') as React.FC<HeaderProps>;
