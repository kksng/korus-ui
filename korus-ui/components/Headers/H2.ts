import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H2 = htmlTagFactory('H2') as React.FC<HeaderProps>;
