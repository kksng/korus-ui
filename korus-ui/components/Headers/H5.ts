import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H5 = htmlTagFactory('H5') as React.FC<HeaderProps>;
