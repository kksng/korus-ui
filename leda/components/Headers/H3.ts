import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H3 = htmlTagFactory('H3') as React.FC<HeaderProps>;
