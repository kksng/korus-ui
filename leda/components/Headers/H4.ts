import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H4 = htmlTagFactory('H4') as React.FC<HeaderProps>;
