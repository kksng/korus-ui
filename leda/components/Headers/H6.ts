import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H6 = htmlTagFactory('H6') as React.FC<HeaderProps>;
