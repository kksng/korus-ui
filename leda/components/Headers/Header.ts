import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const Header = htmlTagFactory('Header') as React.FC<HeaderProps>;
