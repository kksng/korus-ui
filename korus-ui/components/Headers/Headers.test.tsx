// @ts-nocheck
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Header } from './Header';
import { H6 } from './H6';
import { H5 } from './H5';
import { H4 } from './H4';
import { H3 } from './H3';
import { H2 } from './H2';
import { H1 } from './H1';

describe('Headers SNAPSHOTS', () => {
  it('should render H1', () => {
    const wrapper = shallow(<H1>HEADER</H1>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render H2', () => {
    const wrapper = shallow(<H2>HEADER</H2>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render H3', () => {
    const wrapper = shallow(<H3>HEADER</H3>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render H4', () => {
    const wrapper = shallow(<H4>HEADER</H4>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render H5', () => {
    const wrapper = shallow(<H5>HEADER</H5>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render H6', () => {
    const wrapper = shallow(<H6>HEADER</H6>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Header', () => {
    const wrapper = shallow(<Header>HEADER</Header>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Header with classnames .header and .box', () => {
    const wrapper = shallow(<Header _box _header>HEADER</Header>);

    expect(wrapper.text()).toEqual('HEADER');

    expect(wrapper.find('header').hasClass('box')).toBeTruthy();

    expect(wrapper.find('header').hasClass('header')).toBeTruthy();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
