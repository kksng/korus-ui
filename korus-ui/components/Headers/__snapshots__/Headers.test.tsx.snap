// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers SNAPSHOTS should render H1 1`] = `
<h1>
  HEADER
</h1>
`;

exports[`Headers SNAPSHOTS should render H2 1`] = `
<h2>
  HEADER
</h2>
`;

exports[`Headers SNAPSHOTS should render H3 1`] = `
<h3>
  HEADER
</h3>
`;

exports[`Headers SNAPSHOTS should render H4 1`] = `
<h4>
  HEADER
</h4>
`;

exports[`Headers SNAPSHOTS should render H5 1`] = `
<h5>
  HEADER
</h5>
`;

exports[`Headers SNAPSHOTS should render H6 1`] = `
<h6>
  HEADER
</h6>
`;

exports[`Headers SNAPSHOTS should render Header 1`] = `
<header>
  HEADER
</header>
`;

exports[`Headers SNAPSHOTS should render Header with classnames .header and .box 1`] = `
<header
  className="box header"
>
  HEADER
</header>
`;
